2021-09-04 23:32:19,563 ERROR: Exception on /index [GET] [in /Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py:1454]
Traceback (most recent call last):
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/ozanunal/Desktop/Python/flask_app/app/routes.py", line 38, in index
    return render_template('index.html', title='Home', form=form,
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/templating.py", line 148, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 1068, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 997, in get_template
    return self._load_template(name, globals)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 958, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/loaders.py", line 137, in load
    code = environment.compile(source, name, filename)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 757, in compile
    self.handle_exception(source=source_hint)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/ozanunal/Desktop/Python/flask_app/app/templates/index.html", line 20, in template
    {% endif %}
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endif'. Jinja was looking for the following tags: 'endfor' or 'else'. The innermost block that needs to be closed is 'for'.
2021-09-04 23:32:21,109 ERROR: Exception on /explore [GET] [in /Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py:1454]
Traceback (most recent call last):
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/ozanunal/Desktop/Python/flask_app/app/routes.py", line 158, in explore
    return render_template("index.html", title='Explore', posts=posts.items,
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/templating.py", line 148, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 1068, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 997, in get_template
    return self._load_template(name, globals)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 958, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/loaders.py", line 137, in load
    code = environment.compile(source, name, filename)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 757, in compile
    self.handle_exception(source=source_hint)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/ozanunal/Desktop/Python/flask_app/app/templates/index.html", line 20, in template
    {% endif %}
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endif'. Jinja was looking for the following tags: 'endfor' or 'else'. The innermost block that needs to be closed is 'for'.
2021-09-04 23:32:51,512 INFO: Microblog startup [in /Users/ozanunal/Desktop/Python/flask_app/app/__init__.py:32]
2021-09-04 23:32:54,079 ERROR: Exception on /index [GET] [in /Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py:1454]
Traceback (most recent call last):
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/ozanunal/Desktop/Python/flask_app/app/routes.py", line 38, in index
    return render_template('index.html', title='Home', form=form,
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/flask/templating.py", line 148, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 1068, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 997, in get_template
    return self._load_template(name, globals)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 958, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/loaders.py", line 137, in load
    code = environment.compile(source, name, filename)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 757, in compile
    self.handle_exception(source=source_hint)
  File "/Users/ozanunal/Desktop/Python/flask_app/flaskwork/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/ozanunal/Desktop/Python/flask_app/app/templates/index.html", line 20, in template
    {% endif %}
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endif'. Jinja was looking for the following tags: 'endfor' or 'else'. The innermost block that needs to be closed is 'for'.
